design/pipeline-resources.md

```diff
@@ -27,73 +27,63 @@ resources:
-    type: enum  # type of the pipeline source like azurePipelines, Jenkins etc. 
-    connection: string  # service connection to connect to the source
-    project: string # project for the source; optional for current project  
+    connection: string  # service connection for pipelines in other Azure DevOps organizations or Servers
+    project: string # project for the source; optional for current project
     source: string  # source defintion of the pipeline
     version: string  # version to pick the artifact, optional; defaults to Latest
     branch: string  # branch to pick the artiafct, optional; defaults to master branch
     tags: string # picks the artifacts on from the pipeline with given tag, optional; defaults to no tags
 ```
 
-The inputs starting from `project` to `tags` are provider `type` specific properties. The above schema is for the `type`: `azurePipelines`. The inputs may change for other provider types like Jenkins etc.
-
-
 ### Examples
 
 If you need to consume artifacts from another azure pipeline from the current project and if you dont require setting branch, version and tags etc., this can be shortened to:
 
 ```yaml
-resources:         
+resources:
   pipelines:
-  - pipeline: SmartHotel      
-    source: SmartHotel-CI  # name of the pipeline source definition
+  - pipeline: SmartHotel
+    source: SmartHotel-CI # name of the pipeline source definition
 ```
 
-By default, the `type` is taken as `azurePipelines` unless specified otherwise.
-
-
-In case you need to consume an azure pipeline from other project, then you need to include the project name while providing source name.
+In case you need to consume a Pipeline from other project, then you need to include the project name while providing source name.
 
 ```yaml
-resources:         
+resources:
   pipelines:
-  - pipeline: SmartHotel   
+  - pipeline: SmartHotel
     project: DevOpsProject
-    source: SmartHotel-CI  
+    source: SmartHotel-CI
     branch: releases/M142
 ```
 
-### `downloadArtifact` for pipelines
-
-Artifacts from the `pipeline` resource are automatically downloaded and made available for all the jobs. However, in any of the jobs, you can choose to override and download only specific artifacts using `downloadArtifact` shortcut. Once you use `downloadArtifact` to override and download a specific artifact, automatic artifact download behavior is removed and you need to specify all the artifacts you intend to download in the job.
+### `download` for pipelines
 
+All artifacts from the current pipeline and from all `pipeline` resources are automatically downloaded and made available at the beginning of each job. You can override this behavior: see [Pipeline Artifacts](pipeline-artifacts.md#default-and-named-artifacts) for more details.
 
 ```yaml
 - job: deploy_windows_x86_agent
   steps:
-  - downloadArtifact: SmartHotel   # pipeline resource identifier.
+  - download: SmartHotel   # pipeline resource identifier.
```

Comment:
Does the metadata for the referenced resource go down in the job message, to the agent? We can expose the metadata as variables or, even better so we dont bloat the variables, as expression context.

What is the longer term plan here?

Reply:
Exposing artifact meta-data as variables is one of the value proposition of Resources and it enables bunch of important scenarios. This need not be considered as bloating of variables list. Today, in RM, we expose bunch of build artifact meta-data as variables like source definition, build number, commit sha and few more. Customers use this info as part of their deployment workflow, conditional tasks and triggers.

Reply:
I think everyone's in violent agreement that we need to do this :) The question is the mechanism. It would be nice to hear if there are customer scenarios where auto-injected variables are *required*, or if we can put this into expression context and let customers *opt in* to variable injection.

Reply:
Closing this discussion here for now. Please feel free to continue it in a dedicated Pipeline Resources PR.